proto/snpb/replicator_test.go

```diff
@@ -2,10 +2,74 @@ package snpb
 
 import (
 	"testing"
+	"unsafe"
 
 	"github.com/stretchr/testify/require"
+
+	"github.com/kakao/varlog/pkg/types"
 )
 
+func getPointers[S ~[]E, E any](elems S) []uintptr {
+	ptrs := make([]uintptr, len(elems))
+	for i := range elems {
+		p := uintptr(unsafe.Pointer(&elems[i]))
+		ptrs[i] = p
+	}
+	return ptrs
+}
+
+func TestReplicateRequest(t *testing.T) {
+	reqLen1 := ReplicateRequest{
+		LLSN: []types.LLSN{1},
+		Data: [][]byte{[]byte("one")},
+	}
+	data1, err := reqLen1.Marshal()
+	require.NoError(t, err)
+	_ = data1
+
+	reqLen2 := ReplicateRequest{
+		LLSN: []types.LLSN{1, 2},
+		Data: [][]byte{[]byte("one"), []byte("two")},
+	}
+	data2, err := reqLen2.Marshal()
+	require.NoError(t, err)
+	_ = data2
+
+	t.Run("EqualLength", func(t *testing.T) {
+		var req ReplicateRequest
+		err := req.Unmarshal(data1)
+		require.NoError(t, err)
+		want1 := getPointers(req.LLSN)
+		want2 := getPointers(req.Data)
+
+		req.ResetReuse()
+		err = req.Unmarshal(data1)
+		require.NoError(t, err)
+		got1 := getPointers(req.LLSN)
+		got2 := getPointers(req.Data)
+
+		require.Equal(t, want1, got1)
+		require.Equal(t, want2, got2)
+	})
+
+	t.Run("GrowLength", func(t *testing.T) {
```

Comment:
We need to check whether unmarshal or grow was successful.

Reply:
Done.